DQM/L1TMonitorClient/src/L1TStage2CaloLayer2DEClient.cc

```diff
@@ -1,48 +1,424 @@
+    1008, -0.5, 1007.5);
+  MHTComp_ = ibooker.book1D(
+    "MHTRatio","Data/Emul of MHT; MHT iE_{T}; Events",
+    4096, -0.5, 4095.5);
+  METPhiComp_ = ibooker.book1D(
+    "MHTPhiRatio","Data/Emul of MHT #phi; MHTHF i#phi; Events",
+    1008, -0.5, 1007.5);
+  MHTHFComp_ = ibooker.book1D(
+    "MHTHFRatio","Data/Emul of MHTHF; MHTHF iE_{T}; Events",
+    4096, -0.5, 4095.5);
+  MHTPhiComp_ = ibooker.book1D(
+    "MHTHFPhiRatio","Data/Emul of MHTHF #phi; MHTHF i#phi; Events",
+    1008, -0.5, 1007.5);
+  ETTComp_ = ibooker.book1D(
+    "ETTRatio","Data/Emul of ET Total; ETT iE_{T}; Events",
+    4096, -0.5, 4095.5);
+  ETTEMComp_ = ibooker.book1D(
+    "ETTEMRatio","Data/Emul of ET Total EM; ETTEM iE_{T}; Events",
+    4096, -0.5, 4095.5);
+  HTTComp_ = ibooker.book1D(
+    "HTTRatio","Data/Emul of HT Total; HT iE_{T}; Events",
+    4096, -0.5, 4095.5);
+
+  MinBiasHFP0Comp_ = ibooker.book1D(
+    "MinBiasHFP0Ratio", "Data/Emul MinBiasHFP0; N_{towers}; Events",
+    16, -0.5, 15.5);
+  MinBiasHFM0Comp_ = ibooker.book1D(
+    "MinBiasHFM0Ratio", "Data/Emul MinBiasHFM0; N_{towers}; Events",
+    16, -0.5, 15.5);
+  MinBiasHFP1Comp_ = ibooker.book1D(
+    "MinBiasHFP1Ratio", "Data/Emul MinBiasHFP1; N_{towers}; Events",
+    16, -0.5, 15.5);
+  MinBiasHFM1Comp_ = ibooker.book1D(
+    "MinBiasHFM1Ratio", "Data/Emul MinBiasHFM1; N_{towers}; Events",
+    16, -0.5, 15.5);
+
+  TowerCountComp_ = ibooker.book1D(
+    "TowCountRatio", "Data/Emul Tower Count; N_{towers}; Events",
+    5904, -0.5, 5903.5);
 }
 
 void L1TStage2CaloLayer2DEClient::processHistograms(DQMStore::IGetter &igetter){
-  
+
+  // get reference to relevant summary MonitorElement instances
+  // - high level summary
+  // - eg agreement summary
+  // - energy sum agreement summary
+  // - jet agreement summary
+  // - tau agreement summary
+
+  TH1F * hist;
+  TH1F * newHist;
+
+  MonitorElement * hlSummary_ = igetter.get(
+    "L1TEMU/L1TdeStage2CaloLayer2/expert/CaloL2 Object Agreement Summary");
+  MonitorElement * jetSummary_ = igetter.get(
+    "L1TEMU/L1TdeStage2CaloLayer2/expert/Jet Agreement Summary");
+  MonitorElement * egSummary_ = igetter.get(
+    "L1TEMU/L1TdeStage2CaloLayer2/expert/EG Agreement Summary");
+  MonitorElement * tauSummary_ = igetter.get(
+    "L1TEMU/L1TdeStage2CaloLayer2/expert/Tau Agreement Summary");
+  MonitorElement * sumSummary_ = igetter.get(
+    "L1TEMU/L1TdeStage2CaloLayer2/expert/Emergy Sum Agreement Summary");
+
+  // check for existance of object
+  if (hlSummary_) {
+
+    // reference the histogram in MonitorElement
+    hist = hlSummary_->getTH1F();
+    newHist = hlSummary->getTH1F();
+
+    double totalEvents = 0, goodEvents = 0, totalJets = 0, goodJets = 0,
+      totalEg = 0, goodEg = 0, totalTau = 0, goodTau = 0, totalSums = 0,
+      goodSums = 0;
+
+    double evtRatio = 0, jetRatio = 0, egRatio = 0, tauRatio = 0, sumRatio = 0;
+
+    totalEvents = hist->GetBinContent(1);
```

Comment:
please reformat this to combine the declaration and setting of each variable

double totalEvents = hist->GetBinContent(1);